libs/core/datastructures/include/hpx/datastructures/in_value_result.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace hpx::experimental { namespace ranges {
+    template <typename Iter, typename T>
+    struct in_value_result
+    {
+        [[no_unique_address]] Iter in;
+        [[no_unique_address]] T value;
+
+        template <class I2, class T2>
+        // hpx::convertible?
+        constexpr operator in_value_result<I2, T2>() const&
+        {
+            return {in, value};
+        }
+
+        template <class I2, class T2>
+        // hpx::convertible?
+        constexpr operator in_value_result<I2, T2>() &&
+        {
+            return {std::move(in), std::move(value)};
+        }
+    };
+}}    // namespace hpx::experimental::ranges
```

Comment:
We do already have a list of similar types defined here: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/algorithms/include/hpx/parallel/util/result_types.hpp

Reply:
https://en.cppreference.com/w/cpp/algorithm/ranges/fold_left_with_iter

The examples here seem to suggest that the member variables are to be named as `in` and `value`